client/utils/src/mpsc.rs

```diff
@@ -184,8 +172,8 @@ mod inner {
 					break
 				}
 
-				match self.try_next() {
-					Ok(Some(..)) => count += 1,
+				match self.try_recv() {
+					Ok(_) => count += 1,
```

Comment:
Since the channel supports `len()`, the receiver is not `Clone` and this takes `&mut self` couldn't this whole loop just be simplified to `let count = self.len()`?

Reply:
Yes, makes sence, done.